pkg/provider/server.go

```diff
@@ -0,0 +1,149 @@
+	errorCode               string
+	springCloudConfigClient ConfigClient
+	objects                 []*v1alpha1.ObjectVersion
+	files                   []*v1alpha1.File
+}
+
+type Attributes struct {
+	ServerAddress string `json:"server_address,omitempty"`
+	Application   string `json:"application,omitempty"`
+	Profile       string `json:"profile,omitempty"`
+	FileType      string `json:"file_type,omitempty"`
+}
+
+func (a *Attributes) verify() error {
+	if a.ServerAddress == "" {
+		return fmt.Errorf("serverAddress is not set")
+	}
+
+	if a.Application == "" {
+		return fmt.Errorf("application is not set")
+	}
+
+	if a.Profile == "" {
+		return fmt.Errorf("profile is not set")
+	}
+
+	if a.FileType == "" {
+		return fmt.Errorf("FileType is not set")
+	}
+
+	return nil
+}
+
+// NewSpringCloudConfigCSIProviderServer returns CSI provider that uses the spring as the secret backend
+func NewSpringCloudConfigCSIProviderServer(socketPath string) (*SpringCloudConfigCSIProviderServer, error) {
+	client := NewSpringCloudConfigClient()
+	server := grpc.NewServer()
+	s := &SpringCloudConfigCSIProviderServer{
+		springCloudConfigClient: &client,
+		grpcServer:              server,
+		socketPath:              socketPath,
+	}
+	v1alpha1.RegisterCSIDriverProviderServer(server, s)
+	return s, nil
+}
+
+func (m *SpringCloudConfigCSIProviderServer) Start() error {
+	var err error
+	m.listener, err = net.Listen("unix", m.socketPath)
+	if err != nil {
+		return err
+	}
+	go m.grpcServer.Serve(m.listener)
+	return nil
+}
+
+func (m *SpringCloudConfigCSIProviderServer) Stop() {
+	m.grpcServer.GracefulStop()
+}
+
+// Mount implements a provider csi-provider method
+func (m *SpringCloudConfigCSIProviderServer) Mount(ctx context.Context, req *v1alpha1.MountRequest) (*v1alpha1.MountResponse, error) {
+	var attrib Attributes
+	var secret map[string]string
+	var filePermission os.FileMode
+	var err error
+
+	if m.returnErr != nil {
+		return &v1alpha1.MountResponse{}, m.returnErr
+	}
+	if err = json.Unmarshal([]byte(req.GetAttributes()), &attrib); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal attributes, error: %+v", err)
+	}
+	if err = json.Unmarshal([]byte(req.GetSecrets()), &secret); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal secrets, error: %+v", err)
+	}
+	if err = json.Unmarshal([]byte(req.GetPermission()), &filePermission); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal file permission, error: %+v", err)
+	}
+	if len(req.GetTargetPath()) == 0 {
+		return nil, fmt.Errorf("missing target path")
+	}
+
+	err = attrib.verify()
+	if err != nil {
+		return nil, err
+	}
+
+	fileName := fmt.Sprintf("%s-%s.%s", attrib.Application, attrib.Profile, attrib.FileType)
+	content, err := m.springCloudConfigClient.GetConfig(attrib)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve secrets for %s: %w", fileName, err)
+	}
+	defer content.Close()
+
+	out := &v1alpha1.MountResponse{
+		ObjectVersion: m.objects,
```

Comment:
Similar to my comment regarding `m.files`, it looks like the slice `m.objects` is always empty. Does it make sense to use it here?